### PR TITLE
Command line --preprocess missing some macros. 

### LINKED ISSUE #4152: Command line --preprocess misses data/core/units.cfg (LESS_NIMBLE_ELF, etc)
The logic in wesnoth.cpp's handle_preprocess_command assumes that all of the core WML #defines will be in data/core/macros or data/core/terrain-graphics. The assumption isn't quite true, which means it fails to process add-ons that use `{LESS_NIMBLE_ELF}` (a common define for elvish fighter units).

Of the files that aren't included, only data/core/units.cfg looks likely to be necessary.

```
data/core/hotkeys.cfg:#define IF_APPLE_CMD_ELSE_CTRL
data/core/units.cfg:#define LESS_NIMBLE_ELF
data/core/units.cfg:#define RACIAL_NOTES_ORCS_AND_GOBLINS
data/core/units.cfg:#define WOODLAND_RESISTANCE
data/core/units.cfg:#define FLY_MOVE
data/core/units.cfg:#define FLY_DEFENSE DEFENSE_NUMBER
data/core/units.cfg:#define FLY_RESISTANCE
data/core/units.cfg:#define MOUNTAIN_MOVE
data/core/units.cfg:#define MOUNTAIN_DEFENSE
data/core/units.cfg:#define DRAKEFLY_MOVE
data/core/units.cfg:#define DRAKEFLY_RESISTANCE
data/core/units/humans/Mage_Elder.cfg:#define ELDER_MAGE_LIGHTNING DIRECTION_NUMBER
data/core/units/undead/Corpse_Soulless.cfg:#define UNIT_BODY_SOULLESS_STATS MOVETYPE_ID MOVES_NUMBER HP_AMOUNT
data/core/units/undead/Corpse_Soulless.cfg:#define UNIT_BODY_SOULLESS_GRAPHICS BASE_NAME
data/core/units/undead/Corpse_Walking.cfg:#define UNIT_BODY_WALKING_CORPSE_STATS MOVTYPE_NAME MOVES_NUMBER HP_AMOUNT
data/core/units/undead/Corpse_Walking.cfg:#define UNIT_BODY_WALKING_CORPSE_GRAPHICS BASE_NAME
```

### PR DESCRIPTION
**Pull request by Pentarctagon**
Command line --preprocess doesn't process macros in data/core/units.cfg.

Fixes #4152.

### CODE REVIEW

**Comment by AI0867** [comment]
Given that the relevant tag is `[movetype]` and that the plural would be `movetypes`, I would prefer if the filename was `movetypes.cfg`, not `movestypes.cfg`.

Also, the commit message says nothing about what it is fixing. Issues are external to the repository.

**Comment by Pentarctagon** [comment]
Updated - that was a typo on my part.

File: data/core/macros/movetypes.cfg [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# a special macro to define slightly less nimble elves such
+# as fighters, who don't do quite so well in forest
+#define LESS_NIMBLE_ELF
+    [defense]
+        forest=40
+    [/defense]
+#enddef
+
+#define RACIAL_NOTES_ORCS_AND_GOBLINS
```
Comment (stevecotton) [inline data/core/macros/movetypes.cfg]: This one probably doesn't need to move out of units.cfg. However, if it does move here then the file needs a `#textdomain wesnoth-help`.

Comment (Pentarctagon) [reply data/core/macros/movetypes.cfg]: Is there a preference either way? If not, I'll move it back.
@ProditorMagnus as well.

Comment (ProditorMagnus) [reply data/core/macros/movetypes.cfg]: Not used in ageless, does not matter for me.

Comment (Pentarctagon) [reply data/core/macros/movetypes.cfg]: Alright, I've moved `RACIAL_NOTES_ORCS_AND_GOBLINS` back into `data/core/units.cfg` in that case.

**Pull request merged**